src/CodeCoverage/Collector.php

```diff
@@ -20,20 +25,66 @@ class Collector
 	/**
 	 * Starts gathering the information for code coverage.
 	 * @param  string
+	 * @param  string self::STRATEGY_*
 	 * @return void
 	 */
-	public static function start($file)
+	public static function start($file, $strategy = self::STRATEGY_AUTO)
 	{
-		if (!extension_loaded('xdebug')) {
-			throw new \Exception('Code coverage functionality requires Xdebug extension.');
-		} elseif (self::$file) {
+		if (self::$file) {
 			throw new \LogicException('Code coverage collector has been already started.');
 		}
 
+		$method = "start$strategy";
+		if (!method_exists(__CLASS__, $method)) {
+			throw new \LogicException("Unknown strategy '$strategy'");
+		}
+
 		self::$file = fopen($file, 'a+');
+		self::$method();
+	}
+
+
+	/**
+	 * @return void
+	 */
+	private static function startAuto()
+	{
+		if (PHP_SAPI === 'phpdbg') {
+			self::startPhpDbg();
+		} else {
+			self::startXdebug();
+		}
+	}
+
+
+	/**
+	 * @return void
+	 */
+	private static function startXdebug()
+	{
+		if (!extension_loaded('xdebug')) {
```

Comment:
Why? Phpdbg is pointless when you use xdebux strategy. Did you meant to use or or and?